vllm/model_executor/models/molmo.py

```diff
@@ -0,0 +1,1130 @@
+        quant_config: Optional[QuantizationConfig] = None,
+    ) -> None:
+        super().__init__()
+        # Attention block.
+        self.self_attn = MolmoAttention(config, cache_config, quant_config)
+
+        # MLP block.
+        self.mlp = MolmoMLP(config, quant_config=quant_config)
+
+        # LayerNorm
+        assert config.layer_norm_type == "rms"
+        self.input_layernorm = RMSNorm(config.hidden_size,
+                                       eps=config.layer_norm_eps)
+        self.post_attention_layernorm = RMSNorm(config.hidden_size,
+                                                eps=config.layer_norm_eps)
+        self.norm_after = config.norm_after
+
+    def forward(
+        self,
+        positions: torch.Tensor,
+        hidden_states: torch.Tensor,
+        kv_cache: torch.Tensor,
+        attn_metadata: AttentionMetadata,
+        residual: Optional[torch.Tensor],
+    ) -> Tuple[torch.Tensor, Optional[Tuple[torch.Tensor, torch.Tensor]]]:
+        # Self Attention
+        if self.norm_after:
+            residual = hidden_states
+        elif residual is None:
+            residual = hidden_states
+            hidden_states = self.input_layernorm(hidden_states)
+        else:
+            hidden_states, residual = self.input_layernorm(
+                hidden_states, residual)
+        hidden_states = self.self_attn(
+            positions=positions,
+            hidden_states=hidden_states,
+            kv_cache=kv_cache,
+            attn_metadata=attn_metadata,
+        )
+
+        if self.norm_after:
+            hidden_states = self.input_layernorm(hidden_states)
+            hidden_states = hidden_states + residual
+            residual = hidden_states
+
+        if not self.norm_after:
+            hidden_states, residual = self.post_attention_layernorm(
+                hidden_states, residual)
+        hidden_states = self.mlp(hidden_states)
+        if self.norm_after:
+            hidden_states = self.post_attention_layernorm(hidden_states)
+            hidden_states = hidden_states + residual
+            residual = None
+        return hidden_states, residual
+
+
+class MolmoVisionBackbone(nn.Module):
+    def __init__(
+        self,
+        config: PretrainedConfig,
+        vision_config: VisionBackboneConfig,
+        quant_config: Optional[QuantizationConfig] = None,
+    ) -> None:
+        super().__init__()
+        # TODO: vit_layers hard-coded for now. Consider making it configurable.
+        self.vit_layers = [-2, -9]
+        self.image_num_patch = vision_config.image_num_patch
+        self.llm_patches_per_crop = (
+            (self.image_num_patch[0] + 1) // 2,
+            (self.image_num_patch[1] + 1) // 2,
+        )
+        self.image_vit = VisionTransformer(vision_config, quant_config=quant_config)
+        self.num_prefix_tokens = self.image_vit.num_prefix_tokens
+        assert self.num_prefix_tokens in {0, 1}, "Only 0 or 1 prefix tokens are supported"
```

Comment:
It seems that `self.image_vit.num_prefix_tokens` is always going to be 1? Is this configurable?